probe-rs/src/architecture/riscv/communication_interface.rs

```diff
@@ -396,6 +396,11 @@ impl RiscvCommunicationInterface {
         // Reset error bits from previous connections
         self.dtm.reset()?;
 
+        // enable the debug module
```

Comment:
In multihart systems it is important to initialize the DM before interacting with it, not sure why it works on single hart systems.